tests/phpunit/tests/block-bindings/wpBlockBindingsRegistry.php

```diff
@@ -321,4 +346,46 @@ public function test_is_registered_for_known_source() {
 		$result = $this->registry->is_registered( self::$test_source_name );
 		$this->assertTrue( $result );
 	}
+
+	/**
+	 * Test merging `uses_context` from multiple sources.
+	 *
+	 * @ticket 60525
+	 *
+	 * @covers ::register_block_bindings_source
+	 */
+	public function test_merging_uses_context_from_multiple_sources() {
+		$get_value_callback = function () {
+			return 'Anything';
+		};
+
+		register_block_bindings_source(
+			'test/source-one',
+			array(
+				'label'              => 'Test Source One',
+				'get_value_callback' => $get_value_callback,
+				'uses_context'       => array( 'commonContext', 'sourceOneContext' ),
+			)
+		);
+
+		register_block_bindings_source(
+			'test/source-two',
+			array(
+				'label'              => 'Test Source Two',
+				'get_value_callback' => $get_value_callback,
+				'uses_context'       => array( 'commonContext', 'sourceTwoContext' ),
+			)
+		);
+
+		$block_registry          = WP_Block_Type_Registry::get_instance();
+		$paragraph_block_context = $block_registry->get_registered( 'core/paragraph' )->uses_context;
+		// Check that the resulting `uses_context` contains the values from both sources.
+		$this->assertContains( 'commonContext', $paragraph_block_context );
+		$this->assertContains( 'sourceOneContext', $paragraph_block_context );
+		$this->assertContains( 'sourceTwoContext', $paragraph_block_context );
+		// Check that the resulting `uses_context` has unique values.
+		$this->assertCount( count( array_unique( $paragraph_block_context ) ), $paragraph_block_context );
```

Comment:
It would be better to use a custom block that has a predictable `uses_context` (does not change over time) so we could hardcode the expected count.

Reply:
Ok, we need to use one of the supported blocks. Still, I would put a number rather than compute it on the fly, which makes the test harder to follow.

Reply:
I thought about that, but the problem is that the list of supported blocks, where the source `uses_context` is added, is hardcoded right now. So I don't know how to do it.

Reply:
Another option is to run `$this->assertCount()` before registering every source to show how the number grows.

Reply:
I've hardcoded the expected increment between the original context and the new one in [this commit](https://github.com/WordPress/wordpress-develop/pull/6079/commits/03e66ee954843b1e9e12710fab0a8f93cba95c1b). Let me know if you feel that is easier to understand.